File: monsoon3.0/src/main/java/wtf/monsoon/api/module/Category.java
package wtf.monsoon.api.module;

import lombok.AllArgsConstructor;
import lombok.Getter;

@AllArgsConstructor
public enum Category {

    /**
     * Modules in the combat category
     * E.G. Aura, TargetStrafe
     */
    COMBAT("A"),

    /**
     * Modules in the movement category
     * E.G. Flight, Speed
     */
    MOVEMENT("B"),

    /**
     * Modules in the player category
     * E.G. AutoArmor, Scaffold
     */
    PLAYER("C"),

    /**
     * Modules in the visual category
     * E.G. ESP, Tracers
     */
    VISUAL("D"),

    /**
     * Modules in the exploit category
     * E.G. Disabler, NoC03
     */
    EXPLOIT("E"),

    /**
     * Modules in the ghost category
     * E.G. AutoBridger, ClickDelayRemover
     */
    GHOST("F"),

    /**
     * Modules in the HUD category
     * E.G. ArrayList, Speedometer
     */
    HUD("G"),

    /**
     * Modules in the script category
     * These are added through the scripting system
     */
    SCRIPT("H");

    @Getter
    String icon;
}

File: monsoon-3.x-v2/src/main/java/wtf/monsoon/backend/Category.kt
package wtf.monsoon.backend

/**
 * Not inside the `module` package because this could be used in the future
 * for other features, e.g. category dependent commands
 * @author surge
 * @since 09/02/2023
 */
enum class Category(val iconCode: Char) {

    COMBAT('A'),
    MOVEMENT('B'),
    PLAYER('D'),
    VISUAL('C'),
    EXPLOIT('E'),
    GHOST('F'),
    CLIENT('G'),
    HUD('G'),
    SCRIPT('H')

}